test/e2e/framework/volume_util.go

```diff
@@ -397,13 +397,43 @@ func VolumeTestCleanup(f *Framework, config VolumeTestConfig) {
 	}
 }
 
-// Start a client pod using given VolumeSource (exported by startVolumeServer())
+// TestVolumeClient start a client pod using given VolumeSource (exported by startVolumeServer())
 // and check that the pod sees expected data, e.g. from the server pod.
 // Multiple VolumeTests can be specified to mount multiple volumes to a single
 // pod.
 func TestVolumeClient(client clientset.Interface, config VolumeTestConfig, fsGroup *int64, fsType string, tests []VolumeTest) {
-	By(fmt.Sprint("starting ", config.Prefix, " client"))
+	By(fmt.Sprint("starting ", config.Prefix, "-client"))
 	var gracePeriod int64 = 1
+	var commands []string
+	var fileCheckCommand string
+	var securityContext *v1.PodSecurityContext
+	var nodeSelector map[string]string
+	var image string
+	if !NodeOSDistroIs("windows") {
```

Comment:
can we wrap some of these methods into util functions instead of having to replace them in every test case?

Like: `getSecurityContext`, `getTestImage`, `generateWriteCmd`, `generateReadCommand`, etc.

Reply:
For getSecurityContext, getTestImage etc, I think it works well to put into a util function. But for generate different commands it is hard because there are many different commands used in the test and hard to generalize into a few functions.

Reply:
For the commands, is there a general pattern that the tests are doing?

Like if the test is trying to write a file, can we have a general `writeCmd(filename, content)` method?

Reply:
Yes, I made a few utility functions to address this issue. Thanks!